### DISCUSSION TITLE
App close event

### DISCUSSION THREAD

**Discussion by ygoe** [unoplatform/uno#19010] [Q&A]
### Current behavior

There is no event to know when the app has been closed on Android. [This page](https://platform.uno/docs/articles/android-activities.html) looks like it once tried to explain it, but as with all other documentation about Uno I have seen so far, it is wrong or outdated (can't distinguish the two as a new user).

### Expected behavior

I need to know when the user is no longer using my app so it can disconnect connections and be able to terminate. Now, my app keeps running in the background, keeping the Bluetooth connection that it uses open.

### How to reproduce it (as minimally and precisely as possible)

1. Create an Uno application for Windows 11 and Android
2. Start it on Android using the VS debugger
3. Tap the back button or switch to another app

The debugging will never end.

If it does, then continue to connect to a Bluetooth device and regularly poll data from it in a timer. Maybe it's that what keeps it alive. Anyway, I would stop this and disconnect if I knew that the user has gone away.

### Affected platforms

Android

### IDE

Visual Studio 2022

### IDE version

17.12.3

### Relevant plugins

None

**Comment by jeromelaban** [comment]
Thanks for the question. 

Having an event on app closing is generally not possible, or at best not reliable because the system may close the app forcibly. 

You may be able to know that the app has been suspended by using the `Application.Suspending` event, yet similarly, it may not be called if the system decides to kill the app.

**Reply by ygoe** [reply]
I'll try the suspending thing. Having the app closed completely is fine because then it's not running anymore, which is what I want to avoid. I've seen it keeping the Bluetooth connection open and polling constantly while not being used. While it doesn't hurt, it's not efficient either. (But then again, what is... Everything feels so slow that efficiency is probably not a consideration with .NET/Uno on Android.)
